Edit mode - Front-end

hcarneiro opened the pull request:
This is still a work in progress. @squallstar Will take over this branch to continue the work.

- further ref https://github.com/Fliplet/fliplet-studio/issues/1306

squallstar commented on interface.html:
```diff
@@ -162,6 +162,25 @@ <h3 class="panel-title">Are you sure you want to load a new form?</h3>
+                  </label>
+                </div>
+                <div class="checkbox checkbox-icon">
+                  <input type="checkbox" id="generate_email" name="whenSubimitted" v-model="settings.onSubmit" value="generateEmail">
```
whenSubimitted (typo)

squallstar replied:
also, I don't think the name is needed

squallstar commented on js/libs/builder.js:
```diff
@@ -94,7 +95,9 @@ var app = new Vue({
       redirect: formSettings.redirect,
       toggleTemplatedEmail: formSettings.onSubmit.indexOf('templatedEmail') > -1,
       toggleGenerateEmail: formSettings.onSubmit.indexOf('generateEmail') > -1,
-      showDataSource: formSettings.onSubmit.indexOf('templatedEmail') > -1 || formSettings.onSubmit.indexOf('dataSource') > -1,
```
wouldn't this break existing forms?

hcarneiro replied:
Nope, tested it.

squallstar commented on js/libs/form.js:
```diff
@@ -215,6 +220,8 @@ Fliplet.Widget.instance('form-builder', function(data) {
 
         Fliplet.Navigator.onOffline(function() {
           $vm.isOffline = true;
+          console.log(data);
```
console.log

squallstar commented on js/libs/form.js:
```diff
@@ -215,6 +220,8 @@ Fliplet.Widget.instance('form-builder', function(data) {
         Fliplet.Navigator.onOffline(function() {
           $vm.isOffline = true;
+          console.log(data);
+          $vm.isOfflineMessage = data.dataStore === 'editDataSource' ? 'The data can only be updated when connected to the internet.' : 'This form can only be submitted when connected to the internet.';
```
worth having this on new lines:

```
a = b
  ? c
  : d;
```

hcarneiro commented on js/libs/form.js:
```diff
@@ -222,7 +222,9 @@ Fliplet.Widget.instance('form-builder', function(data) {
-          $vm.isOfflineMessage = data.dataStore && data.dataStore.indexOf('editDataSource') > -1 ? 'The data can only be updated when connected to the internet.' : 'This form can only be submitted when connected to the internet.';
+          $vm.isOfflineMessage = data.dataStore && data.dataStore.indexOf('editDataSource') > -1 ?
+            'The data can only be updated when connected to the internet.' :
+            'This form can only be submitted when connected to the internet.';
```
@squallstar My linter doesn't agree with your suggested way. So I kept the linter recommendation!

squallstar commented:
Picking this up

hcarneiro commented:
@squallstar Fixed conflicts.

hcarneiro commented:
@tonytlwu @squallstar I have tried the following to test the feature:

1. Created a data source with a few columns see link: https://cl.ly/3S3O362L2t28
2. Set up a screen with the Verification: Email component (`fliplet-widget-email-verification`) that link to the data source created
3. Created a form:
    - Added the same fields to match the data source data
    - Changed the `field-id` of each form field to match the column name
    - Selected these two options: https://cl.ly/250U2i1N3940
    - Selected the data source created in step 1
4. Login in using the email in the data source
5. Went to the form and:
    - the form didn't show data on the fields
    - tried filling in some data hoping it would at least change the entry, it submitted, but did nothing to the data source

So unless I'm testing this the wrong way, I guess the feature isn't working!

hcarneiro commented:
I think the problem is that the session is not storing the entry: https://cl.ly/1h232O0e2Y2j
